### PR TITLE
feat(movies): associate locations with movies

### PR DESCRIPTION
**Pull request by paarig**
**What:** Implements a POST route, controller method, and a related validator that provides the ability to POST locations where movies are filmed using the /movies/{movie_id}/locations endpoint and get location information for each movie when performing GET requests to the /movies endpoint.

**Why:** This provides the client with the ability to use the API to add and get information about locations where movies are filmed.

**Details:**
Creates a route to allow the client to send POST requests to the /movies/{movie_id}/locations endpoint endpoint.
Includes a controller method to add location info about a specific movie to the DB.
Includes a validator that validates the location info provided by the user when making the POST request.
Implements unit tests and integration tests to ensure the proper functioning of the controller, validator, the /movies endpoint, and the /movies/{movie_id}/locations endpoint as a whole.

### CODE REVIEW

**Comment by sjl2** [comment]
Can you update your PR summary with the proper commit message?

File: db/migrations/20190327151952_create_movies.js [hunk 1]
```diff
@@ -5,9 +5,17 @@ exports.up = async (Knex) => {
     table.increments('id').primary();
     table.text('title').notNullable();
     table.integer('release_year');
+  }).createTable('locations', (table) => {
```
Comment (sjl2) [inline db/migrations/20190327151952_create_movies.js]: While this works locally, in production, this change would not be run again. Can you move this functionality to its own separate migration?

File: lib/plugins/features/movies/controller.js [hunk 2]
```diff
@@ -2,14 +2,25 @@
...
 const Movie = require('../../../models/movie');
 
-exports.create = async (payload) => {
+const Location = require('../../../models/location');
+
+exports.createMovie = async (payload) => {
   const movie = await new Movie().save(payload);
 
-  return new Movie({ id: movie.id }).fetch();
+  return new Movie({ id: movie.id }).fetch({ withRelated: ['locations'] });
```
Comment (sjl2) [inline lib/plugins/features/movies/controller.js]: super nit: there are a few locations where the `{ withRelated: ['locations'] }` is used in this file. My recommendation would be to move it to a `const` at the top. 

```js
const MOVIE_FETCH_OPTIONS = {
  withRelated: ['locations']
};

// ...

return new Movie({ id: movie.id }).fetch(MOVIE_FETCH_OPTIONS);
```

File: lib/plugins/features/movies/controller.js [hunk 3]
```diff
@@ -2,14 +2,25 @@
...
+    location = await new Location().save(payload);
+  }
+  await new Movie({ id: movieId }).locations().attach(location);
+  return new Movie({ id: movieId }).fetch({ withRelated: ['locations'] });
 };
 
-exports.find = (params) => {
-  return new Movie()
+exports.find = async (params) => {
+  const movie = await new Movie()
```
Comment (sjl2) [inline lib/plugins/features/movies/controller.js]: nit: Can you leave this as `return`? The function can still be left as `async` to highlight that it returns a promise.

**Comment by paarig** [comment]
@sjl2 Changed the PR title.

**Pull request merged**